report/report_generator.go

```diff
@@ -0,0 +1,185 @@
+		buffer.WriteString("\t To fix, host all services on separate infrastructure\n\n")
+	}
+
+	if len(report.ExifImages) > 0 {
+		if len(report.ExifImages) > 10 {
+			buffer.WriteString("\033[091mHigh Risk:\033[0m Large number of images with EXIF metadata were discovered!\n")
+		} else {
+			buffer.WriteString("\033[091mMedium Risk:\033[0m Small number of images with EXIF metadata were discovered!\n")
+		}
+
+		buffer.WriteString("\t Why this is bad: EXIF metadata can itself deanonymize a user or\n\t service operator (e.g. GPS location, Name etc.). Or, when combined, can be used to link anonymous identities together.\n")
+		buffer.WriteString("\t To fix, re-encode all images to strip EXIF and other metadata.\n")
+		buffer.WriteString("\t Images Identified:\n")
+		for _, image := range report.ExifImages {
+			buffer.WriteString(fmt.Sprintf("\t\t%s\n", image.Location))
+		}
+		buffer.WriteString("\n")
+	}
+
+	if len(report.OpenDirectories) > 0 {
+		if len(report.OpenDirectories) > 10 {
+			buffer.WriteString("\033[091mMedium Risk:\033[0m Large number of open directories were discovered!\n")
+		} else {
+			buffer.WriteString("\033[091mLow Risk:\033[0m Small number of open directories were discovered!\n")
+		}
+
+		buffer.WriteString("\t Why this is bad: Open directories can reveal the existence of files\n\t not linked from the sites source code. Most of the time this is benign, but sometimes operators forget to clean up more sensitive folders.\n")
+		buffer.WriteString("\t To fix, use .htaccess rules or equivalent to make reading directories listings forbidden.\n")
+		buffer.WriteString("\t Directories Identified:\n")
+		for _, dir := range report.OpenDirectories {
+			buffer.WriteString(fmt.Sprintf("\t\t%s\n", dir))
+		}
+		buffer.WriteString("\n")
+	}
+
+	if report.ResponseHeaders != nil {
+		if _, ok := report.ResponseHeaders["X-FRAME-OPTIONS"]; !ok {
+			buffer.WriteString("\033[091mLow Risk:\033[0m missing X-Frame-Options HTTP header discovered!\n")
+			buffer.WriteString("\t Why this is bad: Provides Clickjacking protection. Values: deny - no rendering within a frame, sameorigin - no rendering if origin mismatch, allow-from: DOMAIN - allow rendering if framed by frame loaded from DOMAIN\n")
+			buffer.WriteString("\t To fix, use X-Frame-Options: deny\n")
+		}
+		if _, ok := report.ResponseHeaders["X-XSS-PROTECTION"]; !ok {
+			buffer.WriteString("\033[091mLow Risk:\033[0m missing X-XSS-Protection HTTP header discovered!\n")
+			buffer.WriteString("\t Why this is bad: his header enables the Cross-site scripting (XSS) filter built into most recent web browsers. It's usually enabled by default anyway, so the role of this header is to re-enable the filter for this particular website if it was disabled by the user. This header is supported in IE 8+, and in Chrome (not sure which versions). The anti-XSS filter was added in Chrome 4. Its unknown if that version honored this header.\n")
```

Comment:
[T]his header

Also, I think we should drop the lines about IE 8 and Chrome. (And does TBB respect this header)?